modules/storages/spec/common/storages/peripherals/storage_interaction/one_drive/files_query_spec.rb

```diff
@@ -68,38 +71,67 @@
+                                  location: "/All Japan Revenues By City.xlsx",
                                   permissions: nil })
   end
 
   it 'when the argument folder is nil, gets information from that users root folder' do
-    stub_request(:get, "https://graph.microsoft.com/v1.0/drives/#{storage.drive_id}/root/children?$select=id,name,size,webUrl,lastModifiedBy,createdBy,fileSystemInfo,file,folder,parentReference")
+    stub_request(
+      :get,
+      "https://graph.microsoft.com/v1.0/drives/#{storage.drive_id}/root/children#{described_class::FIELDS}"
+    )
       .with(headers: { 'Authorization' => "Bearer #{token.access_token}" })
       .to_return(status: 200, body: json, headers: {})
 
     storage_files = described_class.call(storage:, user:, folder: nil).result
 
-    expect(storage_files.size).to eq(38)
+    expect(storage_files.files.size).to eq(38)
+    expect(storage_files.parent.name).to eq("Root")
+    expect(storage_files.ancestors).to be_empty
   end
 
   context 'when accessing a specific folder' do
     let(:json) { read_json('specific_folder') }
 
     it 'uses the specific drive url' do
-      uri = "https://graph.microsoft.com/v1.0/drives/#{storage.drive_id}/items/01BYE5RZYJ43UXGBP23BBIFPISHHMCDTOY/children?$select=id,name,size,webUrl,lastModifiedBy,createdBy,fileSystemInfo,file,folder,parentReference"
+      uri = "https://graph.microsoft.com/v1.0/drives/#{storage.drive_id}/items/01BYE5RZYJ43UXGBP23BBIFPISHHMCDTOY/children#{described_class::FIELDS}"
       stub_request(:get, uri)
         .with(headers: { 'Authorization' => "Bearer #{token.access_token}" })
         .to_return(status: 200, body: json, headers: {})
 
       storage_files = described_class.call(storage:, user:, folder: '01BYE5RZYJ43UXGBP23BBIFPISHHMCDTOY').result
 
-      expect(storage_files.size).to eq(22)
+      expect(storage_files.files.size).to eq(22)
+      expect(storage_files.parent.name).to eq('Class Documents')
+
+      expect(storage_files.ancestors.size).to eq(1)
+    end
+  end
+
+  context 'when accessing a folder two levels deep in the hierarchy' do
+    let(:payload) { read_json('two_levels_deep_folder') }
+    let(:storage) do
+      create(:one_drive_storage,
+             :with_oauth_client,
+             drive_id: 'b!-RIj2DuyvEyV1T4NlOaMHk8XkS_I8MdFlUCq1BlcjgmhRfAj3-Z8RY2VpuvV_tpd')
+    end
+
+    it 'has one ancestors and one parent' do
+      stub_request(:get, "https://graph.microsoft.com/v1.0/drives/#{storage.drive_id}/items/01BYE5RZ2LCUJQWLOZYNDLDC4DRRGJ7OEN/children#{described_class::FIELDS}")
+        .with(headers: { 'Authorization' => "Bearer #{token.access_token}" })
+        .to_return(status: 200, body: payload, headers: {})
+
+      storage_files = described_class.call(storage:, user:, folder: "01BYE5RZ2LCUJQWLOZYNDLDC4DRRGJ7OEN").result
+
+      expect(storage_files.ancestors.size).to eq(2)
+      expect(storage_files.ancestors.map(&:location)).to contain_exactly("/", "/Contoso Clothing")
```

Comment:
❓ ah, so the decision for now is to cut everything before the colon and just return the location on the assumed (and configured drive)?

🟢 I think it'll work. Fine with this approach.

Reply:
It keeps the values similar to what we had for nextcloud. =)